torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -8820,4 +8821,108 @@ def celu_hook(module, _input, output):
             self.assertEqual(ddp_grads[0], local_model.fc.weight.grad)
             self.assertEqual(ddp_grads[1], local_model.fc.bias.grad)
 
+        def _test_hook_pickling(self, hook, hook_state):
+            learning_rate = 0.01
+            chkpt_file = tempfile.gettempdir() + "/checkpoint.pt"
+            rank = self.rank
+
+            input = torch.randn(1, 1, device=rank)
+            net = torch.nn.Linear(1, 5).to(rank)
+            ddp_model = torch.nn.parallel.DistributedDataParallel(
+                net,
+                device_ids=[rank]
+            )
+            dummy_ddp_model = torch.nn.parallel.DistributedDataParallel(
+                net,
+                device_ids=[rank]
+            )
+            optimizer = torch.optim.SGD(ddp_model.parameters(), lr=learning_rate)
+            ddp_model.register_comm_hook(hook_state, hook)
+
+            for _ in range(10):
+                optimizer.zero_grad()
+                loss = ddp_model(input).sum()
+                loss.backward()
+                optimizer.step()
+
+            state = {
+                'state_dict': ddp_model.state_dict(),
+                'comm_hook': pickle.dumps(hook),
+                'comm_hook_state': pickle.dumps(hook_state)
+            }
+
+            if rank == 0:
+                torch.save(state, chkpt_file)
+
+            dist.barrier()
+            map_location = {'cuda:%d' % 0: 'cuda:%d' % rank}
+            checkpoint = torch.load(chkpt_file, map_location=map_location)
+
+            dummy_ddp_model.load_state_dict(checkpoint['state_dict'])
+            dummy_hook = pickle.loads(checkpoint['comm_hook'])
```

Comment:
Similar to the comment with saving. I wonder if we can just do `checkpoint = torch.load(...)` and set `dummy_hook = checkpoint['comm_hook']` here (and similar for `dummy_hook_state`).

Reply:
Updated. Left a comment regarding this earlier.